docs/Wiki/ProductAndTestProjects.md

```diff
@@ -1,29 +1,60 @@
-﻿This page refers to the SonarScanner for .NET, also known as SonarScanner for MSBuild (S4MSB). You can read more about how to install and use it in the ["SonarScanner for .NET documentation"](https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html).
+﻿This page refers to the SonarScanner for .NET (S4NET), also known as SonarScanner for MSBuild (S4MSB). You can read more about how to install and use it in the ["SonarScanner for .NET documentation"](https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html).
+
+### Analyzer references
+
+* the `SonarAnalyzer.CSharp`/`SonarAnalyzer.VB` analyzers will be added to the build even if those analyzers are not referenced in the MSBuild project (S4NET will add references to those analyzers on the fly during the build). The set of rules that is executed is determined by the project type and Quality Profile for the Sonar project in SonarQube/SonarCloud.
+* any third-party analyzers referenced by the MSBuild project will be executed as part of the build (e.g. via NuGet package references). Issues from those analyzers will be uploaded to SonarQube/SonarCloud as [External Issues](https://sonarcloud.io/documentation/analysis/external-issues/). You can configure the third-party analyzer rules as normal using a `Ruleset`. During the build, S4NET will merge your custom ruleset with a ruleset generated from the Quality Profile. In the event of a conflict, the settings in the generated ruleset take precedence.
 
 ### Differences between analysis of test projects and product projects
 
-The Sonar Scanner for MSBuild ("S4MSB") analyses MSBuild projects containing product code differently from projects containing test code.
+The S4NET analyses MSBuild projects containing product code differently from projects containing test code.
+
+#### Analysis of product projects
+
+* analysis rules will be run against product projects, and the issues raised will be uploaded to SonarQube/SonarCloud.
+* only rules related to product code will execute.
+* metrics and Lines of Code ("LOC") limit for commercial versions of SonarQube or private SonarCloud projects are calculated.
 
 #### Analysis of test projects
-* analysis rules are not run against test projects i.e. no issues will be reported to SonarQube/SonarCloud. This is the case even if the test project references third-party NuGet analyzer packages - those analyzers will not be executed.
+
+* analysis rules will be run against test projects, unless excluded (see below), and the issues raised will be uploaded to SonarQube/SonarCloud.
+* only rules related to test code will execute.
 * test projects do not count towards to the Lines of Code ("LOC") limit for commercial versions of SonarQube or private SonarCloud projects.
```

Comment:
Is it still correct ?

Reply:
Yes, test projects are still not included in LOC to avoid sudden licensing surprises on client side

Reply:
Ok, is that a Scanner engine behavior ? I wasn't aware of that

Reply:
Yes, it's on Scanner side. See this:
https://github.com/SonarSource/sonar-scanner-msbuild/blob/0779f4d58eec352021ce36e32fe8df574c78bba8/src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets#L353-L365

Until recently, analyzer didn't know what Scanner considered as a test code. So Analyzer created the `metrics.pb` file containing LOC and other stats for every project. And this target file then overwrites the output binary file to zero-length file (hacky).

In the next release of analyzer, we already can detect a test project from Scanner side and we'll not generate the file to save performance.

Reply:
Line 359 and 364 do the trick in the link above